src/metabase/pulse/render/body.clj

```diff
@@ -549,23 +549,53 @@
      |--------- max-width = (- 59 0) = 59 ---------------------|
 
   overlap = (/ overlap-width max-width) = (/ 35 59) = 0.59"
-  [col-a col-b]
-  (let [[min-a min-b]    (map #(get-in % [:fingerprint :type :type/Number :min]) [col-a col-b])
-        [max-a max-b]    (map #(get-in % [:fingerprint :type :type/Number :max]) [col-a col-b])
-        valid-ranges?    (and min-a min-b max-a max-b
-                              ;; ranges with same min and max won't be considered ranges.
-                              (not= min-a max-a)
-                              (not= min-b max-b))
+  [vals-a vals-b]
+  (let [[min-a max-a]          (-> vals-a sort ((juxt first last)))
+        [min-b max-b]          (-> vals-b sort ((juxt first last)))
+        valid-ranges?          (and min-a min-b max-a max-b
+                                    ;; ranges with same min and max won't be considered ranges.
+                                    (not= min-a max-a)
+                                    (not= min-b max-b))
         overlapping-and-valid? (and valid-ranges?
                                     (or (<= min-a min-b max-a)
                                         (<= min-a max-b max-a)))]
-    (if
-     overlapping-and-valid?
+    (cond
+      ;; any nils in the ranges means we can't compare them.
+      (some nil? (concat vals-a vals-b)) 0
+
+      overlapping-and-valid?
       (let [[a b c d]     (sort [min-a min-b max-a max-b])
             max-width     (- d a)
             overlap-width (- c b)]
-        (/ overlap-width max-width))
-      0)))
+        (/ (double overlap-width) (double max-width)))
+
+      ;; if either range is just a single point, and it's inside the other range,
+      ;; we consider it overlapped. Not likely in practice, but could happen.
+      (and (= min-a max-a) (<= min-b min-a max-b)) 1
+      (and (= min-b max-b) (<= min-a min-b max-a)) 1
+
+      ;; if we have a range that is a single point, is outside the other range, but
+      ;; is near it, we will calculate a 'nearness', which we will consider an overlap.
```

Comment:
Adding the concept of `nearness` sort of contradicts the name of the function being `overlap` which could be confusing. It's also a pretty long function so might be worth refactoring.

Here's a quick suggestion: split this out into two separate `overlap` and `nearness` functions and have them both called by a more generically-named function like `axis-group-threshold-val`.